src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php

```diff
@@ -80,6 +81,81 @@ public function getLogs()
         return $this->data['logs'] ?? [];
     }
 
+    public function getProcessedLogs()
+    {
+        if (null !== $this->processedLogs) {
+            return $this->processedLogs;
+        }
+
+        $rawLogs = $this->getLogs();
+        if ([] === $rawLogs) {
+            return $this->processedLogs = $rawLogs;
+        }
+
+        $logs = [];
+        foreach ($this->getLogs()->getValue() as $rawLog) {
+            $rawLogData = $rawLog->getValue();
+
+            if ($rawLogData['priority']->getValue() > 300) {
+                $logType = 'error';
+            } elseif (isset($rawLogData['scream']) && false === $rawLogData['scream']->getValue()) {
+                $logType = 'deprecation';
+            } elseif (isset($rawLogData['scream']) && true === $rawLogData['scream']->getValue()) {
+                $logType = 'silenced';
+            } else {
+                $logType = 'regular';
+            }
+
+            $logs[] = [
+                'type' => $logType,
+                'errorCounter' => isset($rawLogData['errorCounter']) ? $rawLogData['errorCounter']->getValue() : 1,
+                'timestamp' => $rawLogData['timestamp']->getValue(),
+                'priority' => $rawLogData['priority']->getValue(),
+                'priorityName' => $rawLogData['priorityName']->getValue(),
+                'channel' => $rawLogData['channel']->getValue(),
+                'message' => $rawLogData['message'],
+                'context' => $rawLogData['context'],
+            ];
+        }
+
+        // sort logs from oldest to newest
+        usort($logs, static function ($logA, $logB) {
```

Comment:
wait isnt the default sorting sufficient? arent logs already tracked in order of execution .. could this actually mangle logs in the same second?

Reply:
I did this because deprecations were displayed at the bottom, although they were the first to generate.

In my tests, log order wasn't modified by this for logs in the second.

Reply:
i figure the root cause is those being appened always last here

```
$this->data['logs'] = $this->sanitizeLogs(array_merge($this->logger->getLogs($this->currentRequest), $containerDeprecationLogs));
```

ie. $containerDeprecationLogs, right?

i have no efficient sorting algorithm at hand momentarily :D but we should be really carefull here without milliseconds.